stor/dx.py

```diff
@@ -99,6 +140,17 @@ def wrapper(self):
     realpath = _noop('realpath')
     expanduser = _noop('expanduser')
 
+    def clear_cached_properties(self):
+        """Clears certain cached properties in DXPath objects.
+
+        The canonical and virtual forms of DXPath objects are cached to not hit
+        the server for every transformation call. However, after copy/remove/rename,
+        the cached information is outdated and needs to be cleared.
+        """
+        for prop in ('canonical_project', 'canonical_resource', 'virtual_path'):
+            if prop in self.__dict__:
```

Comment:
nit: `self.__dict__.pop(prop, None)` is shorter

Reply:
Changed